app/controllers/users_email_controller.rb

```diff
@@ -29,14 +29,18 @@ def index
   end
 
   def update
+    if !SiteSetting.enable_secondary_emails && request.post?
```

Comment:
I don't think we should be using the same route for both `POST` and `PUT`.

Usually, the `POST` is to add/create a new item whereas a `PUT` is to update a specific item (and the route usually has an `id`)